src/enzyme_ad/jax/Passes/EnzymeHLOOpt.cpp

```diff
@@ -6029,6 +6097,41 @@ struct PowSimplify
   }
 };
 
+struct NoNanPowSimplify final
+    : public CheckedOpRewritePattern<stablehlo::PowOp, NoNanPowSimplify> {
+  using CheckedOpRewritePattern<stablehlo::PowOp,
+                                NoNanPowSimplify>::CheckedOpRewritePattern;
+
+  NoNanPowSimplify(bool allowOnFloatingPointMath, MLIRContext *context,
+                   PatternBenefit benefit = 1)
+      : CheckedOpRewritePattern(context, benefit),
+        allowOnFloatingPointMath(allowOnFloatingPointMath) {}
+
+  LogicalResult matchAndRewriteImpl(stablehlo::PowOp op,
+                                    PatternRewriter &rewriter) const {
+    if (!canApplyNoNanPattern(allowOnFloatingPointMath, op.getType())) {
+      return failure();
+    }
+
+    // pow(x, 0) -> 1 || pow(1, x) -> 1
+    if ((matchPattern(op.getRhs(), m_Zero()) ||
```

Comment:
I think my reading of ieee spec is that this does not require no nan actually

Reply:
the pow(0, x) simplification leads to a ~1.5x slowdown atleast on GPUs.

Reply:
I will keep the pattern but remove it from defaults